gtfu/src/main/java/gtfu/Direction.java

```diff
@@ -0,0 +1,65 @@
+package gtfu;
+
+import java.io.Serializable;
+import java.io.IOException;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
+public class Direction implements Serializable {
+    String routeDirectionID;
+    String routeID;
+    String directionID;
+    String name;
+
+    public Direction(String routeDirectionID, String routeID, String directionID, String name) {
+        this();
+
+        this.routeDirectionID = routeDirectionID;
+        this.routeID = routeID;
+        this.directionID = directionID;
+        this.name = name;
+    }
+
+    private Direction() {
```

Comment:
the private constructor and the serialization methods `read()` and `write()` are relics of a time when I thought we would have to constantly run the prediction engine on always-on servers. This thankfully turned out to be false, and thus we shouldn't have these constructs anymore.